--- reroute_email.module	2010-04-12 11:43:04.000000000 +0200
+++ x_reroute_email.module	2010-04-12 11:47:15.000000000 +0200
@@ -65,6 +65,9 @@ function reroute_email_mail_alter(&$mess
   // Prepend to the body of the email
   $message['body'] = array_merge($msg, isset($message['body']) ? (is_array($message['body']) ? $message['body'] : array($message['body'])) : array());
 
+  //12/04/2010 - GeorgeZ - Added the line below to fix this issue: http://drupal.org/node/488032
+  $message['body'] = implode($message['body'], "<br/>");
+
   // Change the $to address to be the one we defined
   $message['to'] = variable_get(REROUTE_EMAIL_ADDRESS, ini_get('sendmail_from'));
 }

